### PR TITLE
hight level kafka consumer support

### PR DESCRIPTION
**Pull request by lgbo-dev**
high level kafka consumer in data_reader

### CODE REVIEW

File: xdl/xdl/data_io/fs/file_system_kafka.cc [hunk 1]
```diff
@@ -33,29 +35,31 @@ namespace io {
 /// Kafka IO ant
 class IOAntKafka: public IOAnt {
  public:
-  IOAntKafka(RdKafka::Topic *topic, RdKafka::Consumer *consumer, int partition) : topic_(topic), consumer_(consumer), partition_(partition), offset_(0) { }
+  IOAntKafka(RdKafka::KafkaConsumer *consumer) : consumer_(consumer) { }
   ~IOAntKafka() { 
-    consumer_->stop(topic_, partition_);
-    delete topic_;
+    consumer_->close();
```
Comment (zhanglistar) [inline xdl/xdl/data_io/fs/file_system_kafka.cc]: 判空

File: xdl/xdl/data_io/fs/file_system_kafka.cc [hunk 2]
```diff
@@ -33,29 +35,31 @@ namespace io {
...
         return 0;
       case RdKafka::ERR__PARTITION_EOF:
         XDL_LOG(WARNING) << "Kafka consume reach end of partition";
         return 0;
-      case RdKafka::ERR_NO_ERROR:
+      case RdKafka::ERR_NO_ERROR:{
         XDL_LOG(DEBUG) << "Kafka consume read " << msg->len();
+        std::string data;
+        data.assign((const char*)msg->payload(), msg->len());
+        XDL_LOG(INFO) << "kafka message:" << data;
```
Comment (zhanglistar) [inline xdl/xdl/data_io/fs/file_system_kafka.cc]: 改debug

File: xdl/xdl/data_io/fs/file_system_kafka.cc [hunk 3]
```diff
@@ -33,29 +35,31 @@ namespace io {
...
         break;
-
+      }
       default:
-        XDL_LOG(ERROR) << "Kafka consume fail! [" << topic_ << "," << partition_ << "]";
+        XDL_LOG(ERROR) << "Kafka consume fail! ";
```
Comment (zhanglistar) [inline xdl/xdl/data_io/fs/file_system_kafka.cc]: 加上topic partition

File: xdl/xdl/data_io/fs/file_system_kafka.cc [hunk 4]
```diff
@@ -168,8 +162,15 @@ size_t FileSystemKafka::Size(const char *path) {
 FileSystemKafka::~FileSystemKafka() { }
 
 FileSystem *FileSystemKafka::Get(const char* namenode) {
-  static std::shared_ptr<FileSystemKafka> inst(new FileSystemKafka(namenode));
-  return inst.get();
+  static std::map<std::string, std::shared_ptr<FileSystemKafka> > insts;
```
Comment (zhanglistar) [inline xdl/xdl/data_io/fs/file_system_kafka.cc:170]: 线程不安全？？？？

Comment (lgbo-dev) [reply xdl/xdl/data_io/fs/file_system_kafka.cc:170]: 可以改成threadlocal

**Pull request merged**